sc-poc/pkg/apis/core/v1alpha1/auth_secret_rsa.go

```diff
@@ -0,0 +1,46 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// JwtRSASecret is the schema for the rsasecret api
+type JwtRSASecret struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   JwtRSASecretSpec   `json:"spec,omitempty"`
+	Status JwtRSASecretStatus `json:"status,omitempty"`
+}
+
+// JwtRSASecretSpec defines the desired state of JwtRSASecretSpec
+type JwtRSASecretSpec struct {
+	AuthSecret `json:",inline"`
+
+	// Value holds the content of the secret or jwk url
+	// +kubebuilder:validation:Optional
+	PublicValue string `json:"public,omitempty"`
+
+	PrivateValue string `json:"private,omitempty"`
```

Comment:
I suggest we make these of type `*SecretSource` instead of `string`. lets add a `Value` in `SecretSource` struct which can correspond to the string value we are using here. Make similar change for HSA
```suggestion
	// PublicKey holds the value of the public key
	PublicKey *SecretSource `json:"publicKey"`

	// PrivateKey holds the value of the private key
	// +kubebuilder:validation:Optional
	PrivateKey *SecretSource `json:"privateKey,omitempty"`
```